ckanext/dcat/tests/test_harvester.py

```diff
@@ -27,13 +27,6 @@
 
 
 
-
-# TODO move to ckanext-harvest
-@pytest.fixture
-def harvest_setup():
-    harvest_model.setup()
```

Comment:
Perhaps is best to keep the concerns for the `clean_db` and `harvest_setup` fixtures separated and keep `harvest_setup`  but refactoring it to use `migrate_db_for('harvest')` rather than `harvest_model.setup()`?

Reply:
We couldn't fix the `problem` even after using ` migrate_db_for('harvest')` in `harvest_setup`, as well as redefining `clean_db` in the harvester specific tests (`test_harvester.py` & `test_json_harvester.py`).
All Tests that include `clean_db` still throw the following exception:

>   sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "harvest_object_extra"  does not exist
...
>  /srv/app/src/ckan/ckan/tests/helpers.py:83: in reset_db
>     model.repo.rebuild_db()
> /srv/app/src/ckan/ckan/model/__init__.py:229: in rebuild_db
>     self.delete_all()
> /srv/app/src/ckan/ckan/model/__init__.py:250: in delete_all
>     connection.execute('delete from "%s"' % table.name)
> /usr/lib/python3.8/site-packages/sqlalchemy/engine/base.py:982: in execute
>     return self._execute_text(object_, multiparams, params)
> /usr/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1149: in _execute_text
>     ret = self._execute_context(
> /usr/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1247: in _execute_context
>     self._handle_dbapi_exception(
> /usr/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1466: in _handle_dbapi_exception
>     util.raise_from_cause(sqlalchemy_exception, exc_info)
> /usr/lib/python3.8/site-packages/sqlalchemy/util/compat.py:399: in raise_from_cause
>     reraise(type(exception), exception, tb=exc_tb, cause=cause)
> /usr/lib/python3.8/site-packages/sqlalchemy/util/compat.py:153: in reraise
>     raise value.with_traceback(tb)
> /usr/lib/python3.8/site-packages/sqlalchemy/engine/base.py:1243: in _execute_context

Any ideas on how to initialize/create the plugin db tables or to put the right scope for harvest_setup ?

Maybe this was also mentioned in this [discussion](https://github.com/ckan/ckan/pull/7523#discussion_r1294292054)